[OUDS] Add draft component table (#3471)
Orange-OpenSource/Orange-Boosted-Bootstrap: site/src/content/docs/components/table.mdx

---
title: Table

> description: Documentation and examples for opt-in styling of tables (given their prevalent use in JavaScript plugins) with OUDS Web.

Reviewer: I do not understand the part in parenthesis here

Reviewer: Ok I understand after reading the Overview, still I think it's not really necessary to have this detail in the subtitle...

aliases:
  - "/docs/content/tables/"
  - "/docs/components/table/"
toc: true
---

import { getData } from '@libs/data'
import { getVersionedDocsPath } from '@libs/path'

<Callout type="negative" title="Draft component">
  The table component is not yet designed, and this component is only a draft for the moment. It will be updated in the future with the final design and will certainly need adjustments regarding the DOM. Use it as a temporary component to speed up your development.

  Please refer to the [Tables in the Boosted documentation](https://boosted.orange.com/docs/content/tables/) to have a full view of the table component and its variations. You’ll only need to adapt the utilities inside the DOM.
</Callout>

## Overview

> Due to the extensive use of `<table>` elements in third-party widgets such as calendars and date pickers, OUDS Web’s tables are **opt-in**. This means that you must add the base class `.table` to any `<table>`, to benefit from OUDS Web's table styles which you can then extend with our optional modifier classes or custom styles. All table styles are not inherited in OUDS Web, meaning any nested tables can be styled independent of the parent.

Reviewer: I see it comes from Bootstrap but it's not very clear, asked a llm to improve it
```suggestion
Due to the extensive use of `<table>` elements in third-party widgets such as calendars and date pickers, OUDS Web’s tables are **opt-in**. This means that you must add the base class `.table` to any `<table>`, to benefit from OUDS Web's table styles which you can then extend with our optional modifier classes or custom styles. In OUDS Web, table styles are not inherited, enabling independent styling for nested tables.
```

          <td>Cell</td>
          <td>Cell</td>
          <td>Cell</td>
          <td>Cell</td>
          <td>Cell</td>
          <td>Cell</td>
          <td>Cell</td>
          <td>Cell</td>
        </tr>
        <tr>
          <th scope="row">3</th>
          <td>Cell</td>
          <td>Cell</td>
          <td>Cell</td>
          <td>Cell</td>
          <td>Cell</td>
          <td>Cell</td>
          <td>Cell</td>
          <td>Cell</td>
          <td>Cell</td>
        </tr>
      </tbody>
    </table>
  </div>`} />

<Code buttonLabel="table responsive" lang="html" code={`
  <div class="table-responsive">
    <table class="table">
      ...
    </table>
  </div>`} />

### Breakpoint specific

Use `.table-responsive{-xs|-sm|-md|-lg|-xl|-2xl|3xl}` as needed to create responsive tables up to a particular breakpoint. From that breakpoint and up, the table will behave normally and not scroll horizontally.

**These tables may appear broken until their responsive styles apply at specific viewport widths.**

<Details summary="Examples of responsive tables up to particular breakpoint">
  <br/>
  Resize window to see differences:
  {getData('breakpoints').map((breakpoint) => {

>     return (

Reviewer 2: Maybe we could add a small title before each table just to tell which breakpoint is use in the following table ?
